### ISSUE TITLE
Memory leak in CheckoutActivity

### ISSUE THREAD

**Issue by umangce** [razorpay/razorpay-android-sample-app#30]
According to android studio, `com.razorpay.CheckoutActivity` has a memory leak or leaks. You code has been obfuscated. So, I can't find where memory leak has happened. 

Can you please look into this? Let me know if you require more information.

Thanks,
Umang Kothari

**Comment by hariramsvnit** [comment]
@umangce Can you give more information about the approach you followed to detect memory leak?

A general approach to detect memory leak is to use the app and monitor memory uses if you see something unusual take Java Heap dump and analyze the heap dump to find the root cause of the leak. I have followed this approach to verify it a few months back but didn't find any.

**Comment by umangce** [comment]
I have followed the same way. I have found this leak on Moto G (1st Gen) which runs on Android Lollipop (5.1). 

I started payment screen (CheckoutActivity) and just cancelled payment screen. Then after I checked memory monitor, dumped Java Heap and found memory leak. Please check attached screen-shot from Android Studio. Unfortunately, I can't share whole dump file.

![screenshot from 2017-06-10 15-49-06](https://user-images.githubusercontent.com/11502033/27002092-c845aac0-4df5-11e7-94e0-7e013e399802.png)

**Comment by hariramsvnit** [comment]
@umangce In the case of memory leak, multiple objects get created and even if you force run GC, the GC won't be able to collect it. Here I can see multiple objects of few classes but after this stage can you force run GC(you will get the option for this in Android Studio) and compare the dump before running GC and after running GC?

**Comment by umangce** [comment]
@hariramsvnit I have captured screen-shot after running GC. Otherwise there is no point to dump Java Heap to check memory leak :)

**Comment by hariramsvnit** [comment]
@umangce Cool, Thanks for reporting the issue, we will look into it.

**Comment by hariramsvnit** [comment]
The issue has been fixed in version [1.4.0](https://github.com/razorpay/razorpay-android-sample-app/releases/tag/v1.4.0)

**Comment by umangce** [comment]
@hariramsvnit I am afraid to say that it still leaks in 1.4.0 :(

**Comment by hariramsvnit** [comment]
@umangce We have used leakcanary(https://github.com/square/leakcanary) to detect memory leaks and we have fixed all which we have found. If you can point to specific memory leak it will be helpful in understanding which memory leak still persists.

**Comment by umangce** [comment]
@hariramsvnit I have checked out your demo which you have put on Github. And it has same memory leak.
![screenshot from 2017-07-20 16-12-35](https://user-images.githubusercontent.com/11502033/28413803-8f799036-6d66-11e7-84fa-f32a46eb397e.png)

I have uploaded same memory dump on Google drive. Download from [here](https://drive.google.com/open?id=0B8G1jtqOU3QDZ1RCa2RCRlphQ1k).

Let me know if you require information.
